netbeans/CheckerFrameworkPlugin/src/org/checkerframework/netbeans/CheckerFrameworkPanel.java

```diff
@@ -0,0 +1,159 @@
+                                            InstalledFileLocator.getDefault()
+                                                    .locate(
+                                                            "checkerstrings.properties",
+                                                            "org.checkerframework.netbeans",
+                                                            false)))
+                            .entrySet();
+        } catch (IOException e) {
+            System.out.println("Failed to load checker strings properties file.");
+        }
+
+        checkerList = new JCheckBox[checkerStrings.size()];
+        int i = 0;
+        selection =
+                new StringBuilder(editableProperty.get("annotation.processing.processors.list"));
+        for (Entry<String, String> e : checkerStrings) {
+            checkerList[i] = new JCheckBox(e.getValue());
+            if (selection.toString().contains(e.getKey())) {
+                checkerList[i].setSelected(true);
+            }
+            this.add(checkerList[i]);
+            checkerList[i].addItemListener(new CheckBoxListener(selection));
+            i++;
+        }
+    }
+
+    /**
+     * Helper method to load a properties file
+     *
+     * @param propsFO File object to be loaded
+     * @return EditableProperties class loaded with the properties file
+     * @throws IOException
+     */
+    private static EditableProperties loadProperties(FileObject propsFO) throws IOException {
+        InputStream propsIS = propsFO.getInputStream();
+        EditableProperties props = new EditableProperties(true);
+        try {
+            props.load(propsIS);
+        } finally {
+            propsIS.close();
+        }
+        return props;
+    }
+
+    /**
+     * This method is called by the action listener that runs when the Project Properties window is
+     * closed and will store the selected checkers into the project.properties file.
+     *
+     * @throws IOException
+     */
+    public void store() throws IOException {
+        try {
+            ProjectManager.mutex()
+                    .writeAccess(
+                            new WriteCheckerFrameworkPropertiesAction(
+                                    projectProperties, checkerPath, checkerQualPath, selection));
+        } catch (MutexException mux) {
+            throw (IOException) mux.getException();
+        }
+    }
+
+    /** ItemListener that updates the checkers to run based on the check boxes that get selected. */
+    private static class CheckBoxListener implements ItemListener {
+        private StringBuilder s;
+        private String selectedChecker;
+
+        public CheckBoxListener(StringBuilder inString) {
+            s = inString;
+        }
+
+        @Override
+        public void itemStateChanged(ItemEvent e) {
+            JCheckBox source = (JCheckBox) e.getItemSelectable();
+
+            for (Entry<String, String> entry : checkerStrings) {
```

Comment:
You create a new CheckBoxListener for each of the check boxes. Here you go through all checkerStrings, to again determine for which of the checkers you created this listener.
You have two ways to make this nicer:
- Use just one Listener for all the checkboxes and then keep this loop to determine which one was selected
- Use distinct Listeners for the distinct checkboxes, but when creating the listener you can also pass the name of the checker for which this is the listener. Then you can remove this list.